framework/src/araui/core/ui_core.c

```diff
@@ -330,6 +325,48 @@ static void _ui_redraw(uint32_t dt)
 	ui_window_redraw_list_clear();
 }
 
+static void _ui_update_redraw_list_recur(ui_widget_body_t *widget)
+{
+	int iter;
+	static bool update_flag = false;
+	ui_widget_body_t *child;
+
+	if ((widget->update_flag || update_flag) && widget->parent) {
+		// Update previous area
+		ui_window_add_redraw_list(widget->global_rect);
+
+		ui_renderer_translate(&widget->parent->trans_mat, &widget->trans_mat, (float)widget->local_rect.x, (float)widget->local_rect.y);
+		ui_renderer_rotate(&widget->trans_mat, widget->degree);
+		ui_renderer_scale(&widget->trans_mat, widget->scale_x, widget->scale_y);
+
+		// Update new area
+		ui_widget_update_global_rect(widget);
+		ui_window_add_redraw_list(widget->global_rect);
+		widget->update_flag = false;
+		update_flag = true;
+	}
+
+	vec_foreach(&widget->children, child, iter) {
```

Comment:
I think you can change this to loop. We have a small memory, we have to reduce stack as possible as we can.

Reply:
Thank you for your opinion. Without recursive function call is much better than current one. I'll modify it.

Reply:
Hmm. I'll try to change this to the loop.